### ISSUE TITLE
[Doc]: Provide a working example for turning on specific axes labels when sharex or sharey are used with subplots

### ISSUE THREAD

**Issue by benoitespinola** [matplotlib/matplotlib#24958]
### Documentation Link

https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.subplots.html

### Problem

I find the feature to turn off the labels for all the subplots with a shared axis that are not in the edge of the subplots' grid surprising and confusing.
The documentation does state this as a wanted feature. And does state that turning the labels back on require one to use tick_params.

However, it is not clear what the user needs to do to turn a specific axis's tick labels back on.

I did not find a single example on how to do this in the entire documentation from Matplotlib.

I would be happy to contribute writing a working example (I just need to know how to do it).

### Suggested improvement

To include a working examples on how to turn a specific axis' tick label back on (one for the x-axis, a second for the y-axis).

Here is the suggested starting point:

```
import numpy as np
import matplotlib.pyplot as plt

n_cols = 4
n_rows = 5

fig, axs = plt.subplots(n_cols, n_rows,  sharex = True,  sharey = True)

for ax in axs.ravel():
    x = np.random.rand(20)
    y = np.random.rand(20)

    ax.scatter(x,y)

# turn on x-axis tick labels at [2,2]
axs[1,1].tick_params(labelbottom=True)

# same for y-axis tick labels at [2,3]
axs[1,2].tick_params(labelleft=True)

# same for both x and y axes tick labels for [3,3]
axs[2,2].tick_params(labelbottom=True, labelleft=True)

plt.show()
```

**Comment by jklymak** [comment]
That seems reasonable.  I'd add it to https://matplotlib.org/devdocs/tutorials/intermediate/arranging_axes.html#sphx-glr-tutorials-intermediate-arranging-axes-py

**Comment by benoitespinola** [comment]
Or maybe here:
https://matplotlib.org/3.1.1/gallery/subplots_axes_and_figures/shared_axis_demo.html

I would also have a grid that is not a 1-d grid to show which index corresponds to rows and which one corresponds to columns in the axes variable (I always get those confused).

**Comment by oscargus** [comment]
I agree that https://matplotlib.org/stable/gallery/subplots_axes_and_figures/shared_axis_demo.html should be updated. It is a bit confusing that it explicitly removes shared axis, so probably it is quite old. Better to show an example that they are removed automatically and then show how to get them back. And a 2 x 3 can illustrate a few different variants, including the indexing.

**Comment by timhoffm** [comment]
Might also be added https://matplotlib.org/stable/gallery/subplots_axes_and_figures/subplots_demo.html#sharing-axes.

I think these two examples should both be deleted as they use a clunky incremental `subplot()` creation, which IMHO is not great generating multiple subplots, in particular if you go into more complex logic like sharing.

- https://matplotlib.org/stable/gallery/subplots_axes_and_figures/shared_axis_demo.html
- https://matplotlib.org/stable/gallery/subplots_axes_and_figures/share_axis_lims_views.html
